paderbox/array/sparse.py

```diff
@@ -0,0 +1,842 @@
+        """
+        >>> pb.utils.pretty.pprint(SparseArray(10))
+        SparseArray(shape=(10,))
+        >>> a = SparseArray(10)
+        >>> a[:5] = 1
+        >>> a[7:] = 2
+        >>> pb.utils.pretty.pprint(a)
+        SparseArray(_SparseSegment(onset=0, array=array([1, 1, 1, 1, 1])),
+                    _SparseSegment(onset=7, array=array([2, 2, 2])),
+                    shape=(10,))
+        >>> a.pad_value=-1
+        >>> pb.utils.pretty.pprint(a)
+        SparseArray(_SparseSegment(onset=0, array=array([1, 1, 1, 1, 1])),
+                    _SparseSegment(onset=7, array=array([2, 2, 2])),
+                    shape=(10,), pad_value=-1)
+        """
+        if cycle:
+            p.text(f'{self.__class__.__name__}(...)')
+        else:
+            name = self.__class__.__name__
+            pre, post = f'{name}(', ')'
+            with p.group(len(pre), pre, post):
+                for idx, m in enumerate(self._segments):
+                    p.pretty(m)
+                    p.text(',')
+                    p.breakable()
+                p.text(f'shape={self.shape}')
+                if self.pad_value != 0:
+                    p.text(f', pad_value={self.pad_value}')
+
+    def __setitem__(self, item, value):
+        """
+        >>> a = SparseArray(10)
+        >>> a[:3] = 1
+        >>> a[3:5] = np.arange(2) + 1
+        >>> a[6:8] = np.ones(2, dtype=int)
+        >>> np.asarray(a)
+        array([1, 1, 1, 1, 2, 0, 1, 1, 0, 0])
+
+        >>> a = SparseArray(10)
+        >>> a[:100] = 1
+        >>> a
+        SparseArray(_SparseSegment(onset=0, array=array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])), shape=(10,))
+
+        # Multi-dimensional
+        >>> a = SparseArray((2, 10))
+        >>> a[..., :5] = 1
+        >>> a[..., 7:] = np.arange(6).reshape((2, 3))
+        >>> np.asarray(a)
+        array([[1, 1, 1, 1, 1, 0, 0, 0, 1, 2],
+               [1, 1, 1, 1, 1, 0, 0, 3, 4, 5]])
+        >>> a[0, 5:] = 2
+        Traceback (most recent call last):
+          ...
+        NotImplementedError: Only ellipsis is allowed for first dimensions in setitem
+
+
+        # Test some exceptions
+        >>> a = SparseArray(shape=10)
+        >>> a[:5] = np.ones(3)
+        Traceback (most recent call last):
+         ...
+        ValueError: Could not assign array with shape (3,) to SparseArray slice with size 5
+        >>> a[:5] = np.ones((1, 5))
+        Traceback (most recent call last):
+         ...
+        ValueError: Shape mismatch: SparseArray has shape (10,), but assigned array has shape (1, 5).
+        >>> a[:100] = np.ones(100)
+        Traceback (most recent call last):
+         ...
+        ValueError: Could not assign array with shape (100,) to SparseArray slice with size 10
+        """
+        if isinstance(item, tuple):
+            selector = item[:len(self.shape) - 1]
+            # Complex indexing is not supported in __setitem__
+            if len(selector) != 1 or selector[0] is not ...:
```

Comment:
What do you want to check here? In `[x, y]` you can use every combination, but in `[x, y, z]`, `x` have to be `...`